flutter_secure_storage_web/lib/flutter_secure_storage_web.dart

```diff
@@ -70,50 +71,46 @@ class FlutterSecureStorageWeb extends FlutterSecureStoragePlatform {
   }) async {
     final map = <String, String>{};
     final prefix = "${options[_publicKey]!}.";
-    for (int j = 0; j < html.window.localStorage.length; j++) {
-      final entry = html.window.localStorage.entries.elementAt(j);
-      if (!entry.key.startsWith(prefix)) {
+    for (int j = 0; j < web.window.localStorage.length; j++) {
+      final key = web.window.localStorage.key(j) ?? "";
+      if (!key.startsWith(prefix)) {
         continue;
       }
 
-      final value = await _decryptValue(entry.value, options);
+      final value = await _decryptValue(web.window.localStorage.getItem(key), options);
 
       if (value == null) {
         continue;
       }
 
-      map[entry.key.substring(prefix.length)] = value;
+      map[key.substring(prefix.length)] = value;
     }
 
     return map;
   }
 
-  crypto.Algorithm _getAlgorithm(Uint8List iv) =>
-      crypto.Algorithm(name: 'AES-GCM', length: 256, iv: iv);
+  js_interop.JSObject _getAlgorithm(Uint8List iv) {
```

Comment:
Specifying `JSAny` instead of `JSObject` may reduce type conversion.

https://github.com/dart-lang/web/blob/v0.5.1/lib/src/dom/webcryptoapi.dart